libs/barista-components/sunburst/src/sunburst.ts

```diff
@@ -0,0 +1,261 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Overlay, OverlayConfig, OverlayRef } from '@angular/cdk/overlay';
+import { TemplatePortal } from '@angular/cdk/portal';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Inject,
+  Input,
+  OnChanges,
+  Optional,
+  Output,
+  SimpleChanges,
+  TemplateRef,
+  ViewChild,
+  ViewContainerRef,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  dtSetUiTestAttribute,
+  DtUiTestConfiguration,
+  DT_UI_TEST_CONFIG,
+} from '@dynatrace/barista-components/core';
+import { DtSunburstOverlay } from './sunburst.directive';
+import {
+  DtSunburstNode,
+  DtSunburstNodeInternal,
+  DtSunburstSlice,
+  DtSunburstValueMode,
+  fillNodes,
+  getNodesWithState,
+  getSelectedId,
+  getSelectedNodes,
+  getSelectedNodesFromOutside,
+  getSlices,
+  getValue,
+} from './sunburst.util';
+
+const OVERLAY_PANEL_CLASS = 'dt-sunburst-overlay-panel';
+
+@Component({
+  selector: 'dt-sunburst',
+  templateUrl: 'sunburst.html',
+  styleUrls: ['sunburst.scss'],
+  preserveWhitespaces: false,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.Emulated,
+})
+export class DtSunburst implements OnChanges {
+  readonly width = 480;
+  readonly viewBox = '-240 -176 480 352';
+
+  @ViewChild('svg') svgEl;
+
+  /** @internal Template reference for the DtSunburst overlay. */
+  @ContentChild(DtSunburstOverlay, { read: TemplateRef })
+  // tslint:disable-next-line: no-any
+  private _overlay: TemplateRef<any>;
+
+  filledSeries: DtSunburstNodeInternal[];
+  slices: DtSunburstSlice[];
+
+  _selected: DtSunburstNodeInternal[];
+  selectedLabel: string;
+  selectedValue: number;
+  selectedRelativeValue: number;
+  labelAsAbsolute: boolean = true;
+
+  @HostListener('click', ['$event']) onClick(ev?: MouseEvent): void {
+    this.select(ev);
+  }
+
+  @Input() series: DtSunburstNode[];
+  @Input() selected: DtSunburstNode[];
+  @Input() noSelectionLabel: string = 'All';
+  @Input() valueDisplayMode: DtSunburstValueMode = DtSunburstValueMode.ABSOLUTE;
+
+  @Output() selectedChange: EventEmitter<DtSunburstNode[]> = new EventEmitter();
+
+  /** Template portal for the default overlay */
+  // tslint:disable-next-line: use-default-type-parameter no-any
+  private _portal: TemplatePortal<any> | null;
+
+  /** Reference to the open overlay. */
+  private _overlayRef: OverlayRef;
+
+  constructor(
+    private _overlayService: Overlay,
+    private _viewContainerRef: ViewContainerRef,
+    private _elementRef?: ElementRef<HTMLElement>,
+    @Optional()
+    @Inject(DT_UI_TEST_CONFIG)
+    private _config?: DtUiTestConfiguration,
+  ) {}
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.series?.currentValue !== changes.series?.previousValue) {
+      this.filledSeries = fillNodes(this.series);
+      this.render();
+    }
```

Comment:
You could do this in the setter of the series data as well. 
Granted you would loose the comparison, but given that this one will only check for instance equality and not deep object equality, it might be better to deal with this on in the setter

```typescript
// Line 91
@Input()
get series(): DtSunburstNode[] {
  return this._series;
}
set series(value: DtSunburstNode[]) {
  this._series = value;
  this.filledSeries = fillNodes(value);
  this.render();
}
```

If you would want to wait for all things to be evaluated before running render, you could next an internal state subject within the ngOnChanges when your if conditions apply and to it only once a cycle. If performance is an issue in this case.